### DISCUSSION TITLE
I PLAY THE MUSIC WITH A  PHONE'S MUSIC PLAYER ,BUT FIND SOME BUGS.

### DISCUSSION THREAD

**Discussion by qzbh** [pschatzmann/ESP32-A2DP#573] [Q&A] [answered]
My mobile phone is connected to esp32  Bluetooth  and then I click to play a music.  It can play normally  ,but once I pause to play again  it doesn't work normally.It will disconnect.If i wang to connect Bluetooth ,I need to click Bluetooth manually on my phone.
My code :

#include "AudioTools.h"
#include "BluetoothA2DPSink.h"

I2SStream i2s;
BluetoothA2DPSink a2dp_sink(i2s);

void setup() {
  auto cfg = i2s.defaultConfig();
  cfg.channels = 2;
  cfg.bits_per_sample = 16;
  cfg.buffer_count = 8;
  cfg.buffer_size = 256;
  cfg.sample_rate = 44100;
  cfg.use_apll = true;
  cfg.pin_bck = 14;
  cfg.pin_ws = 15;
  cfg.pin_data = 22;
    
  i2s.begin(cfg);
  a2dp_sink.start("MyMusic");
}
void loop() {
}

**Comment by pschatzmann** [comment, marked as answer]
I can'r reproduce your problem! 

I was testing today again with the latest ESP32 Core version 3.0.1 and my Android Phone with different applications hitting pause and play of with different durations and it was working perfectly.

What core are you using ?

You can try to analyze what's going on in detail by activating both the A2DP and Audiotools logging. I recommend to use the log level info for both.  The Readme of the corresponding projects are desribing how to do this...

**Reply by qzbh** [reply]
![tb_image_share_1718265563845.jpg.png](https://github.com/pschatzmann/ESP32-A2DP/assets/127125197/a21b6ec6-4354-4a5c-b00a-3779076c0ed4)

**Reply by pschatzmann** [reply]
No, you need to check your Arduino ESP32 core version number that you can see in the Board Manager or your Arduino IDE

**Reply by qzbh** [reply]
My core maybe not the latest version. I will try to check it later.
